lib/internal/test_runner/reporter/spec.js

```diff
@@ -59,6 +63,36 @@ class SpecReporter extends Transform {
       ), `\n${indent}  `);
     return `\n${indent}  ${message}\n`;
   }
+  #reportCoverage(color, nesting, summary, symbol) {
+    let report = `${color}${this.#indent(nesting)}${symbol}start of coverage report${white}\n`;
+    report += `${color}${this.#indent(nesting)}${symbol}file | line % | branch % | funcs % | uncovered lines${white}\n`;
```

Comment:
When I wrote this table for the TAP reporter, I kind of assumed that there is a good chance the TAP will be processed by something else to make it display more nicely. For the spec reporter, I would expect something a little nicer looking out of the box.

Reply:
maybe coverage should have a separate full-blown reporter?

Reply:
No strong feeling there.

Reply:
I don't know how much Node's spec reporter should provide out of the box. But I guess the two common elements for spec type reporters are:
- output statistics formatted into a table
- percentage thresholds indicated by color (green, yellow & red).

Reply:
At https://github.com/nodejs/node/issues/46665#issuecomment-1463961596 and https://github.com/nodejs/node/issues/47032, I suggested since spec reporter is intended to provide an user readable output, I propose to use [Istambul text reporter](https://istanbul.js.org/docs/advanced/alternative-reporters/#text), that's the same being used by [Jest](https://jestjs.io/), [NYC](https://github.com/istanbuljs/nyc), and by extension also [c8](https://github.com/bcoe/c8).

Reply:
@MoLow @cjihrig looks good already (v20.0.0) but is there any chance will see the coverage table formatted as such, with columns aligned? It is very hard to parse and gets messy quickly the greater the number of test files.

Reply:
> @MoLow @cjihrig looks good already (v20.0.0) but is there any chance will see the coverage table formatted as such, with columns aligned? It is very hard to parse and gets messy quickly the greater the number of test files.

Agree, it's difficult to read in its current state, some extra whitespaces would not hurt.

Reply:
a PR is welcomed

Reply:
FWIW, I never intended for the table used by the TAP reporter (which could presumably be parsed by a different tool) to be shared with the more human readable spec reporter.

Reply:
Your draft is fine, maybe try to make it more readable and know where it starts and ends. Personally I really like [IstanbulJS text reporter](https://istanbul.js.org/docs/advanced/alternative-reporters/#text) (Disclaimer: I improved it)

Reply:
Adding the box drawing will waste horizontal space, but table borders can help to identify where the table starts and ends.

Reply:
@dmnsgn your draft looks good. Please keep in mind edge cases that will impact formatting, such as:

1. Very long filenames
2. Files with a long list of uncovered lines.

Reply:
@cjihrig one solution to that I have seen in the wild was:
- truncate filename `...ery-long/file-name.js`
- truncate lines number and group line numbers `...12-15, 20, 23-27`
If that's desirable, we could base it on a table max width.

@piranna how about this:
<img width="500" alt="Screenshot 2023-04-28 at 14 24 16" src="https://user-images.githubusercontent.com/1636460/235159583-90549490-af75-46ca-94b1-e5e1acc47fc9.png">

Reply:
IstanbulJS reporter already support that edge cases.

Reply:
> @piranna how about this:

Looks nice, I like it :-) Just remove the extra whitespace in `line %` and `funcs %` columns to save horizontal space.

Are going filenames and number to have some extra colors too, like yellow and red?

Reply:
Here are some screenshot at different available terminal width:
- Full width
<img width="1486" alt="Screenshot 2023-04-29 at 13 44 25" src="https://user-images.githubusercontent.com/1636460/235303148-1e4b6c7c-c293-4816-8167-8e044a791cf4.png">

- Medium:
<img width="844" alt="Screenshot 2023-04-29 at 13 44 39" src="https://user-images.githubusercontent.com/1636460/235303153-a5f656c3-21b4-45ab-9a2e-c27b81991ca6.png">

- Small:
<img width="461" alt="Screenshot 2023-04-29 at 13 44 49" src="https://user-images.githubusercontent.com/1636460/235303159-893c436b-89d0-4701-bb9c-4ccad7387902.png">

With uncovered lines as ranges and filename with colors.
I will prepare a PR if that's okay.

Reply:
> Here are some screenshot at different available terminal width:

I love it!!! 😍 


> With uncovered lines as ranges and filename with colors.

I would also set yellow for percentages between red and green, and I would remove the comma between the ranges, the whitespace is enough separator. Also in ranges, I would set red for uncovered lines and yellow for uncovered branches, if possible at the same time, then this would be a full replacement for IstanbulJs text reporter :-)

Reply:
@dmnsgn I think it looks fantastic.

> I will prepare a PR if that's okay.

Please do.